### PR TITLE
[CHK-3075] feat: add api v2 for helpdeskController

### PR DESCRIPTION
**Pull request by scaminati-bv**
#### List of Changes

- add `/v2/helpdesk/searchTransaction` api 
- add `HelpdeskService` v2 to retrieve transaction on both mongo-oracle databases
- add `PMTransactionDataProvider` for v2 api to restore oracle db access

#### Motivation and Context

The goal of this PR is to add v2 api also for helpdeskController. 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-spec/v2/openapi.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ paths:
           required: true
           description: "Max element per page"
       tags:
-        - eCommerce
+        - ecommerce
```
Comment (pietro-tota) [inline api-spec/v2/openapi.yaml:36]:
```suggestion
        - eCommerce
```

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/v2/ResponseBuilderV1.kt [hunk 2]
```diff
@@ -33,6 +38,66 @@ fun buildTransactionSearchResponse(
         )
         .transactions(results)
 
+// TODO
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/v2/ResponseBuilderV1.kt:41]: forgotten comment or are we missing something?

Comment (scaminati-bv) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/v2/ResponseBuilderV1.kt:41]: we inserted this TODO for a second check and we forgot to remove it.

**Review by pietro-tota — DISMISSED** [review]
Well done, just one trivial issue 😄 🚀

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/v2/ResponseBuilderV1.kt (hunk 2 above)
Comment (scaminati-bv) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/v2/ResponseBuilderV1.kt]: Resloved [here](https://github.com/pagopa/pagopa-ecommerce-helpdesk-service/pull/124/commits/6545645d7431833e4a8add069ceb31c48f50be93)

**Review by pietro-tota — APPROVED** [review]

**Comment by pietro-tota** [comment]
@scaminati-bv after openapi tag update seems that EcommerceApi cannot be resolved
````

file:///home/vsts/work/1/s/src/main/kotlin/it/pagopa/ecommerce/helpdesk/controllers/v2/EcommerceController.kt:4:54 Unresolved reference: EcommerceApi
````
Can you please fix re-building locally project and checking for right name to use for api?

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**